src/coordinates/uniform_cylindrical.hpp

```diff
@@ -0,0 +1,517 @@
+  // Cylindrical+Spherical Conversions
+  //----------------------------------------
+  // TODO
+
+  //----------------------------------------
+  // Geometric Terms (Find better names for these!)
+  //----------------------------------------
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h2v(const int i) const { return r_c_(i); };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h2f(const int i) const { return Xf<1>(i); };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h31v(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h31f(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh2vd1(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh2fd1(const int i) const { return 1.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh31vd1(const int i) const { return 0.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh31fd1(const int i) const { return 0.0; };
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h32v(const int j) const { return 1.0; }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real h32f(const int j) const { return 1.0; }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh32vd2(const int j) const { return 0.0; }
+
+  KOKKOS_FORCEINLINE_FUNCTION
+  Real dh32fd2(const int j) const { return 0.0; }
+
+  //What is this?
+  KOKKOS_FORCEINLINE_FUNCTION
+  void GetAcc(const int k, const int j, const int i, const Real &rp,
+	      const Real &cosphip, const Real &sinphip, const Real &zp,
+	      Real &acc1, Real &acc2, Real &acc3) const
+  {
+    const Real cosdphi = cosphi_c_(j)*cosphip - sinphi_c_(j)*sinphip; //cos(x3v(k)-phip)
+    const Real sindphi = sinphi_c_(j)*cosphip - cosphi_c_(j)*sinphip; //sin(x3v(k)-phip)
+    //Psi = - 1.0/sqrt(r0^2 + rp^2 - 2r0*rp*cos(phi-phip) + (z-zp)^2)
+    acc1 = (r_c_(i) - rp*cosdphi); //acc1 = dPsi/dr, Psi=-1/dist2
+    acc2 = rp*sindphi;        //acc2 = dPsi/(r*dphi) 
+    acc3 = Xc<3>(k)-zp;   
+  }
+
+  //----------------------------------------
+  // Position to Cell index
+  //----------------------------------------
+  KOKKOS_INLINE_FUNCTION
+  void Xtoijk(const Real &r, const Real &theta, const Real &z, int &i, int &j, int &k) const {
+    i = (nx_[0] != 1) ? static_cast<int>(
+        std::floor((r - xmin_[0]) / dx_[0])) +
+      istart_[0] : istart_[0] ;
+    j = (nx_[1] != 1) ? static_cast<int>(std::floor(
+          (theta - xmin_[1]) / dx_[1])) +
+      istart_[1]
+      : istart_[1];
+    k = (nx_[2] != 1)? static_cast<int>(std::floor(
+          (z - xmin_[2]) / dx_[2])) +
+      istart_[2]
+      : istart_[2];
+  }
+
+  const std::array<Real, 3> &GetXmin() const { return xmin_; }
+  const std::array<int, 3> &GetStartIndex() const { return istart_; }
+  const char *Name() const { return name_; }
+
+ //private:
+  std::array<int, 3> istart_;
+  std::array<Real, 3> xmin_, dx_, area_, nx_;
+  constexpr static const char *name_ = "UniformCylindrical";
+
+  const std::array<Real, 3> &Dx_() const { return dx_; }
+
+  ParArrayND<Real> r_c_, coord_vol_i_, cosphi_c_, sinphi_c_;
```

Comment:
these are all 1D right? You could use `ParArray1D` here probably.

That said, as a first pass, these calculations are only being cached for performance. We could do them without and just compute inline as well. The only reason not to do this is that the trig functions are not transcendental.